packages/server/src/api/controllers/role.ts

```diff
@@ -226,35 +238,20 @@ export async function accessible(ctx: UserCtx<void, AccessibleRolesResponse>) {
   if (!roleId) {
     roleId = roles.BUILTIN_ROLE_IDS.PUBLIC
   }
+  // If a custom role is provided in the header, filter out higher level roles
+  const roleHeader = ctx.header?.[Header.PREVIEW_ROLE] as string
+  const isBuilder = ctx.user && sharedSdk.users.isAdminOrBuilder(ctx.user)
```

Comment:
When would `ctx.user` ever not be present? I thought `UserCtx` guaranteed it would be, throwing an auth error otherwise?

Reply:
Addressed.